cmd/k8s-operator/proxy.go

```diff
@@ -235,7 +235,8 @@ type impersonateRule struct {
 // addImpersonationHeaders adds the appropriate headers to r to impersonate the
 // caller when proxying to the Kubernetes API. It uses the WhoIsResponse stashed
 // in the context by the apiserverProxy.
-func addImpersonationHeaders(r *http.Request) error {
+func addImpersonationHeaders(r *http.Request, log *zap.SugaredLogger) error {
+	log = log.Named(fmt.Sprintf("request from %s", r.RemoteAddr))
```

Comment:
Perhaps `log = log.With("remote", r.RemoteAddr)` to simply add a field instead of renaming the logger?

Reply:
nice 👍🏼